src/github/entities/user.entity.ts

```diff
@@ -0,0 +1,40 @@
+import { Base } from 'src/universal/base.entity';
+import { PullRequest } from 'src/webhook/entities/pull-request.entity';
+import { Repo } from 'src/webhook/entities/repository.entity';
+import { Column, Entity, OneToMany } from 'typeorm';
+import { UserInstallation } from './user-installation.entity';
+@Entity()
+export class User extends Base {
+  @Column({ type: 'varchar', length: 255, nullable: true })
+  name: string;
+
+  @Column({ type: 'varchar', length: 255, nullable: true })
+  phone: string;
+
+  @Column({ type: 'varchar', length: 255, nullable: true })
+  email: string;
+
+  @Column({ type: 'varchar', nullable: true })
+  password: string;
+
+  @Column({ type: 'varchar', length: 20, unique: true })
+  githubUserId: string;
+
+  @Column({ type: 'varchar', length: 255, unique: true, nullable: true })
+  githubUsername: string;
+
+  @Column({ type: 'varchar', length: 255, nullable: true })
+  githubAvatarUrl: string;
+
+  @Column({ type: 'varchar', length: 255, nullable: true })
+  githubAccessToken: string;
+
+  @Column({ type: 'varchar', nullable: true, length: 255 })
+  accessToken: string;
```

Comment:
PullProofAI

Consider adding validation for input fields like `name`, `phone`, `email`, etc., to ensure data integrity and consistency.
```diff
+   @Column({ type: 'varchar', length: 255, nullable: false })
+   name: string;

+   @Column({ type: 'varchar', length: 20, nullable: false })
+   phone: string;

+   @Column({ type: 'varchar', length: 255, nullable: false })
+   email: string;

+   @Column({ type: 'varchar', nullable: false })
+   password: string;

+   @Column({ type: 'varchar', length: 20, unique: true, nullable: false })
+   githubUserId: string;

+   @Column({ type: 'varchar', length: 255, unique: true, nullable: false })
+   githubUsername: string;

+   @Column({ type: 'varchar', length: 255, nullable: false })
+   githubAvatarUrl: string;

+   @Column({ type: 'varchar', length: 255, nullable: false })
+   githubAccessToken: string;

+   @Column({ type: 'varchar', nullable: false, length: 255 })
+   accessToken: string;
```



<!-- This is an auto-generated comment by PullProofAI -->